cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java

```diff
@@ -266,10 +265,7 @@ protected void configure() {
           .build(new TypeLiteral<ManagerFactory<DeploymentInfo, ApplicationWithPrograms>>() { })
       );
 
-      install(new FactoryModuleBuilder()
-                .implement(Store.class, DefaultStore.class)
-                .build(StoreFactory.class)
-      );
+      bind(Store.class).to(DefaultStore.class).in(Scopes.SINGLETON);
```

Comment:
Store is not thread-safe... I think Terence is right and we need to provide a new instance every time.

Sent from my iPhone

> On Mar 27, 2015, at 10:56 PM, Terence Yim <notifications@github.com> wrote:
> 
> In cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java:
> 
> > @@ -266,10 +265,7 @@ protected void configure() {
> >            .build(new TypeLiteral<ManagerFactory<DeploymentInfo, ApplicationWithPrograms>>() { })
> >        );
> >  
> > -      install(new FactoryModuleBuilder()
> > -                .implement(Store.class, DefaultStore.class)
> > -                .build(StoreFactory.class)
> > -      );
> > +      bind(Store.class).to(DefaultStore.class).in(Scopes.SINGLETON);
> This change the behavior. When factory was used, a new instance is created every time upon request. Do you know if Store is safe to be a singleton?
> 
> —
> Reply to this email directly or view it on GitHub.
>